### PR TITLE
fix: support global "_Headers" in Undici 5.26.3

### LINKED ISSUE #72: Losing headers when using Undici in latest Node 18.x
Node.js's update to Undici 5.26.3 caused its `Headers` class to be called `_Headers`:

https://github.com/nodejs/node/pull/50153/files#diff-f516ab824a7722da938a4c7c851520d39731ddeb4f7198dff4e932c5d4f8fdf7

As a result, passing a native Undici Headers object to headers-polyfill (which happens when MSW intercepts a native Node.js `fetch` request with headers) loses the headers' contents.

To reproduce, run the following script in Node.js 18.16.1 (where it outputs the expected headers) and Node 18.18.2 (where it does not).

```js
const hp = require('headers-polyfill');

console.log(hp.headersToString(new hp.Headers(new Headers({'Content-Type': 'application/json'}))));
```

### PR DESCRIPTION
**Pull request by joshkel**
Node.js's update to Undici 5.26.3 caused its `Headers` class to be called `_Headers`:

https://github.com/nodejs/node/pull/50153/files#diff-f516ab824a7722da938a4c7c851520d39731ddeb4f7198dff4e932c5d4f8fdf7

Fixes #72.

### CODE REVIEW

**Comment by joshkel** [comment]
Alternatively, would it be better to check `instanceof` if a global `Headers` class exists? Perhaps something like this?

```ts
    /**
     * @note Cannot necessarily check if the `init` is an instance of the
     * `Headers` because that class may not be defined in Node or jsdom.
     */
    if (
      ['Headers', 'HeadersPolyfill'].includes(init?.constructor.name) ||
      init instanceof Headers ||
      (typeof globalThis.Headers === 'function' && init instanceof globalThis.Headers)
    ) {
```

**Review by kettanaito — APPROVED** [review]
Thanks for fixing this, @joshkel!

File: src/Headers.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export class Headers {
      * because that class is only defined in the browser.
      */
     if (
-      ['Headers', 'HeadersPolyfill'].includes(init?.constructor.name) ||
+      ['Headers', 'HeadersPolyfill', '_Headers'].includes(init?.constructor.name) ||
```
Comment (kettanaito) [inline src/Headers.ts:28]: I like your proposal to check the `instanceof` against the global `Headers` class if it exists. Perhaps let's use it instead?

```js
|| (typeof globalThis.Headers !== 'undefined' && init instanceof globalThis.Headers)
```

Comment (kettanaito) [inline src/Headers.ts:28]: We should also add a test for this, implicitly. Bumping the used Node.js version in the CI would be enough. I recommend adding a matrix of Node.js versions so we test both older versions and the recent ones that introduce this change. 

Let me know if you need help with this.

Comment (joshkel) [reply src/Headers.ts]: @kettanaito Done, I think. I have no real experience working with GitHub Actions, so I don't know if my approach is will work and is the best way of doing it.

**Review by kettanaito — APPROVED** [review]
This looks fantastic. Thank you, @joshkel!

File: src/Headers.ts (hunk 1 above)
Comment (kettanaito) [reply src/Headers.ts]: You did it right 👌

**Comment by kettanaito** [comment]
Welcome to contributors, @joshkel 🎉

**Comment by kettanaito** [comment]
## Released: v4.0.2 🎉

This has been released in v4.0.2!

- 📄 [**Release notes**](https://github.com/mswjs/headers-polyfill/releases/tag/v4.0.2)
- 📦 [npm package](https://www.npmjs.com/package/headers-polyfill/v/4.0.2)

Make sure to always update to the latest version (`npm i headers-polyfill@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Pull request merged**